ArchUnitNET/Fluent/Syntax/Elements/Types/TypeConditionsDefinition.cs

```diff
@@ -348,13 +348,24 @@ ConditionResult Condition(TRuleType ruleType)
                     .Concat(classDiagramAssociation.GetTargetNamespaceIdentifiers(ruleType))
                     .ToList());
 
+
                 var pass = true;
                 var dynamicFailDescription = "does depend on";
+
+                //Prevent failDescriptions like "does depend on X and does depend on X and does depend on Y and does depend on Y
+                var failDescriptionCache = new List<string>();
+
                 foreach (var dependency in ruleType.GetTypeDependencies())
```

Comment:
I wonder:
Instead of filtering afterwards, wouldn't Distinct() operation on the dependencies list or rather a DistinctBy on the dependencies with regard to the dependency.FullName property be easier to read? 
DistinctBy in LINQ can be emulated with: dependency.GroupBy(p => p.FullName).Select(g => g.First()) 

Could you convert the very good testcase you provided (thanks again!) into an automated test? If you don't find the time I can do this as well, but it might take a while. 

I'm happy to see that ArchUnitNet made its way to Zeiss :)

Reply:
Distinct is now implemented and tested.
The result stays the same but is now easier to read.

The automated test will be added later.

Also we want to thank You (all) for this great Open Source Project.
It does help us a lot for our .NET/Java Health Checks.
Especially the PlantUML Support is a huge help for larger projects.